diff --git a/Makefile.in b/Makefile.in
index d82c0ea..0d5c30a 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -91,7 +91,7 @@ znc: $(BIN_OBJS) $(LIBZNC)
 
 $(LIBZNC): $(LIB_OBJS)
 	$(E) Linking $(LIBZNC)...
-	$(Q)$(CXX) $(LDFLAGS) -shared -o $@ $(LIB_OBJS) $(LIBS)
+	$(Q)$(CXX) $(LDFLAGS) -shared -o $@ $(LIB_OBJS) $(LIBS) -Wl,--out-implib=libznc.dll.a
 endif
 
 unittest: $(LIB_OBJS) test/gtest-all.o test/gtest-main.o $(TESTS)
@@ -178,6 +178,7 @@ install: znc $(LIBZNC)
 	find $(DESTDIR)$(DATADIR)/webskins -type d -exec chmod 0755 '{}' \;
 	find $(DESTDIR)$(DATADIR)/webskins -type f -exec chmod 0644 '{}' \;
 	$(INSTALL_PROGRAM) znc $(DESTDIR)$(bindir)
+	$(INSTALL_PROGRAM) libznc.dll.a $(DESTDIR)$(libdir)
 	$(INSTALL_SCRIPT) znc-buildmod $(DESTDIR)$(bindir)
 	$(INSTALL_DATA) $(srcdir)/include/znc/*.h $(DESTDIR)$(includedir)/znc
 	$(INSTALL_DATA) include/znc/zncconfig.h $(DESTDIR)$(includedir)/znc
diff --git a/configure.ac b/configure.ac
index 3a64922..2465e54 100644
--- a/configure.ac
+++ b/configure.ac
@@ -429,7 +429,7 @@ if test -z "$ISCYGWIN" ; then
 	appendMod -fPIC
 else
 	# But cygwin does want most of ZNC in a shared lib
-	LIBZNC="libznc.dll"
+	LIBZNC="cygznc-1.6.dll"
 	LIBZNCDIR="$bindir"
 fi
 
diff --git a/modules/Makefile.in b/modules/Makefile.in
index 0dea7da..bc6c64d 100644
--- a/modules/Makefile.in
+++ b/modules/Makefile.in
@@ -62,7 +62,7 @@ C=
 endif
 
 ifneq "$(LIBZNC)" ""
-LDFLAGS += -L.. -lznc -Wl,-rpath,$(LIBZNCDIR)
+LIBS += -L.. -lznc -Wl,-rpath,$(LIBZNCDIR)
 endif
 
 CLEAN    :=
diff --git a/modules/modperl/Makefile.inc b/modules/modperl/Makefile.inc
index abc938c..303ef38 100644
--- a/modules/modperl/Makefile.inc
+++ b/modules/modperl/Makefile.inc
@@ -44,7 +44,7 @@ modperl_all: modperl/ZNC.$(PERLCEXT_EXT) modperl/swigperlrun.h modperl/functions
 
 modperl/ZNC.$(PERLCEXT_EXT): modperl/ZNC.o Makefile modperl.so
 	$(E) Linking ZNC Perl bindings library...
-	$(Q)$(CXX) $(MODFLAGS) $(LDFLAGS) $(MODLINK) -o $@ $< $(PERL_LD) $(PERLDEPONMOD)
+	$(Q)$(CXX) $(MODFLAGS) $(LDFLAGS) $(MODLINK) -o $@ $< $(PERL_LD) $(PERLDEPONMOD) $(LIBS)
 
 modperl/ZNC.o: modperl/ZNC.cpp Makefile
 	@mkdir -p modperl
diff --git a/modules/modpython/Makefile.inc b/modules/modpython/Makefile.inc
index c0cde99..e41758e 100644
--- a/modules/modpython/Makefile.inc
+++ b/modules/modpython/Makefile.inc
@@ -52,7 +52,7 @@ modpython/_znc_core.o: modpython/_znc_core.cpp Makefile
 
 modpython/_znc_core.$(PYCEXT_EXT): modpython/_znc_core.o Makefile modpython.so
 	$(E) Linking ZNC python bindings library...
-	$(Q)$(CXX) $(MODFLAGS) $(LDFLAGS) $(MODLINK) -o $@ $< $(PY_LDFLAGS) $(PYDEPONMOD)
+	$(Q)$(CXX) $(MODFLAGS) $(LDFLAGS) $(MODLINK) -o $@ $< $(PY_LDFLAGS) $(PYDEPONMOD) $(LIBS)
 
 ifneq "$(SWIG)" ""
 include $(srcdir)/modpython/Makefile.gen
