diff --git a/include/znc/version.h b/include/znc/version.h
index 360280e..3f8b639 100644
--- a/include/znc/version.h
+++ b/include/znc/version.h
@@ -11,6 +11,9 @@
 #define VERSION        (VERSION_MAJOR + VERSION_MINOR / 10.0)
 
 // You can add -DVERSION_EXTRA="stuff" to your CXXFLAGS!
+#ifndef VERSION_EXTRA
+# define VERSION_EXTRA ""
+#endif
 extern const char* ZNC_VERSION_EXTRA;
 
 #endif // !ZNC_VERSION_H
diff --git a/src/version.cpp b/src/version.cpp
index 360280e..3f8b639 100644
--- a/src/version.cpp
+++ b/src/version.cpp
@@ -1,2 +1,2 @@
 #include <znc/version.h>
-const char* ZNC_VERSION_EXTRA = "";
+const char* ZNC_VERSION_EXTRA = VERSION_EXTRA;
